packages/default-theme/components/cms/elements/SwProductGallery.vue

```diff
@@ -0,0 +1,100 @@
+<template>
+  <div class="sw-product-gallery gallery">
+    <template v-if="mediaGallery.length > 0">
+      <slot name="desktop-galery" v-bind="mediaGallery">
+        <div class="gallery__desktop" v-for="image in mediaGallery" :key="image.big.url">
+          <SfImage
+            :src="image.big.url"
+            class="cover-image desktop-only"
+          />
+        </div>
+      </slot>
+    </template>
+    <div class="gallery__mobile mobile-only" v-if="mediaGallery.length > 0">
+      <slot name="mobile-galery" v-bind="mediaGallery">
+        <SfGallery
+          class="gallery-mobile mobile-only"
+          :images="mediaGallery"
+        />
+     </slot>
+    </div>
+  </div>
+</template>
+
+<script>
+import { SfImage, SfGallery } from '@storefront-ui/vue'
+import { getProductMainImageUrl, getProductMediaGallery } from '@shopware-pwa/helpers'
+
+export default {
+  components: { SfImage, SfGallery },
+  props: {
+    product: {
+      type: Object,
+      default: () => ({}),
+      required: true
```

Comment:
the default value is what protects us when the prop is not provided. We should never use the required properties, unless it's completely necessary.
```suggestion
```